### PR TITLE
Gh-470: Migrate user page from old docs

### LINKED ISSUE #470: Migrate Users section from old docs
The v1 Gaffer docs contain [a developer guide page explaining the `User` class](https://gchq.github.io/gaffer-doc/v1docs/getting-started/developer-guide/users.html). This page should be updated and added to the new documentation site, it can be placed under `/development-guide/project-structure/components/`.

### PR DESCRIPTION
**Pull request by GCHQDeveloper314**
# Related Issue

- Resolve #470

### CODE REVIEW

File: docs/development-guide/project-structure/components/user.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+For users of Gaffer, interacting in Java is insecure as you are able to set any authorisations you want to. However, as
+you are writing Java it is assumed that you have access to the `store.properties` file providing the connection details
+(including password) for the cluster where your data is stored (e.g. Accumulo). Therefore, you could bypass Gaffer and
+read all the data directly from disk if you wanted to.
+
+The security layer for Gaffer is currently only enforced by the REST API. We recommend restricting users such that they
+do not have access to the cluster and by only allowing users to connect to Gaffer via the REST API. In the REST API the
+`User` object is constructed via a [`UserFactory`](https://gchq.github.io/Gaffer/uk/gov/gchq/gaffer/rest/factory/UserFactory.html).
+Currently, we only provide one implementation of this, the [`UnknownUserFactory`](https://gchq.github.io/Gaffer/uk/gov/gchq/gaffer/rest/factory/UnknownUserFactory.html).
```
Comment (tb06904) [inline docs/development-guide/project-structure/components/user.md]: Where exactly do you set the `UserFactory` that is in use? I assume its one of the config files?

Comment (GCHQDeveloper314) [reply docs/development-guide/project-structure/components/user.md]: Great question, I'll try and document this. I think it is somewhere in the rest config yes.

Comment (GCHQDeveloper314) [reply docs/development-guide/project-structure/components/user.md]: I've added a line covering this. However this aspect should be covered in more detail elsewhere in the docs. There are quite a few other REST related properties in [common-rest `SystemProperty.java`](https://github.com/gchq/Gaffer/blob/298604ff4696c6f7fb070125f719cb27ebcb0db4/rest-api/common-rest/src/main/java/uk/gov/gchq/gaffer/rest/SystemProperty.java#L34-L44) which are not covered anywhere at all. Perhaps an issue could be created to create a new page to cover these, which could then be linked from this page.

**Review by cn337131 — APPROVED** [review]

File: docs/development-guide/project-structure/components/user.md (hunk 1 above)
Comment (GCHQDeveloper314) [reply docs/development-guide/project-structure/components/user.md]: [New issue created](https://github.com/gchq/gaffer-doc/issues/475) to add more detail on REST properties in general.

**Review by tb06904 — APPROVED** [review]

**Pull request merged**